src/main/java/uk/gov/companieshouse/api/strikeoffobjections/client/OracleQueryClient.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.companieshouse.api.strikeoffobjections.client;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class OracleQueryClient {
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${ORACLE_QUERY_API_URL}")
+    private String oracleQueryApiUrl;
+
+    public Long getCompanyActionCode(String companyNumber) {
+        String getCompanyActionCodeUrl = String.format("%s/company/%s/action-code", oracleQueryApiUrl, companyNumber);
+
+        ResponseEntity<Long> response = restTemplate.getForEntity(getCompanyActionCodeUrl, Long.class);
```

Comment:
what happens if we get a 404 returned?

Reply:
The service errors - it's caught and logged by the controller and an internal server error is thrown. But, that's what lead me on to that issue we have (I think) within the web client, where it doesn't handle these correctly. It's not related to this change but a general problem. Will post something in Slack and then raise a defect.

This PR is fine though.